src/gantt.ts

```diff
@@ -997,17 +996,33 @@ module powerbi.extensibility.visual {
             return children;
         }
 
-        private static downgradeDurationUnit(durationUnit: string): string {
-            let durationUnitTypeIndex = GanttDurationUnitType.indexOf(durationUnit);
-            durationUnit = (durationUnitTypeIndex !== -1 && durationUnit) || "day";
-            // stepDurationTransformationDefault - variable for accuracy calculation of duration transformation
-            // if duration == 0.84 day, we need transform duration to minutes in order to get duration without extra loss
-            for (let i = stepDurationTransformationDefault; i > 0; i--) {
-                if (durationUnitTypeIndex - i > -1) {
-                    durationUnit = GanttDurationUnitType[durationUnitTypeIndex - i];
+        public static getNewUnitByFloorDurationFloor(durationUnitTypeIndex: number, duration: number): string {
+            if (!durationUnitTypeIndex)
+                return GanttDurationUnitType[0];
+
+            switch (durationUnitTypeIndex) {
+                case 3:
```

Comment:
Can we use `GanttDurationUnitType` instead of numbers?
Numbers are getting code hard to understand